File: app/src/main/java/com/maksk993/cryptoportfolio/view/activities/MainActivity.java
package com.maksk993.cryptoportfolio.view.activities;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.MenuItem;

import com.maksk993.cryptoportfolio.R;
import com.maksk993.cryptoportfolio.contract.MainContract;
import com.maksk993.cryptoportfolio.model.models.FindFragmentById;
import com.maksk993.cryptoportfolio.presenter.MainPresenter;
import com.maksk993.cryptoportfolio.view.fragments.*;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity implements MainContract.MainView {
    FragmentManager fragmentManager = getSupportFragmentManager();
    Map<String, Fragment> fragmentMap = new HashMap<>();
    MainContract.MainPresenter presenter = new MainPresenter(this);
    BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.hide();
        }

        fragmentMap.put("History", new HistoryFragment());
        fragmentMap.put("Portfolio", new PortfolioFragment());
        fragmentMap.put("Settings", new SettingsFragment());

        bottomNavigationView = findViewById(R.id.bottom_nav_menu);

        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                int id = item.getItemId();
                int fragmentId;

                if (id == R.id.nav_history) fragmentId = FindFragmentById.getInteger(FindFragmentById.HISTORY);
                else if (id == R.id.nav_portfolio) fragmentId = FindFragmentById.getInteger(FindFragmentById.PORTFOLIO);
                else fragmentId = FindFragmentById.getInteger(FindFragmentById.SETTINGS);

                presenter.onClick(fragmentId);
                return true;
            }
        });
    }

    @Override
    public void startNewFragment(FindFragmentById id) {
        Fragment fragment;
        switch (id){
            case HISTORY:
                fragment = fragmentMap.get("History");
                break;
            case PORTFOLIO:
                fragment = fragmentMap.get("Portfolio");
                break;
            default:
                fragment = fragmentMap.get("Settings");
        }
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.replace(R.id.nav_host_fragment, fragment);
        transaction.commit();
    }
}

File: app/src/main/java/com/maksk993/cryptoportfolio/presentation/screens/main/MainActivity.kt
package com.maksk993.cryptoportfolio.presentation.screens.main

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.maksk993.cryptoportfolio.R
import com.maksk993.cryptoportfolio.databinding.ActivityMainBinding
import com.maksk993.cryptoportfolio.presentation.models.FindFragmentById
import dagger.hilt.android.AndroidEntryPoint


@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    private val viewModel: MainViewModel by viewModels()
    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        viewModel.apply {
            setSavedAppTheme()
            getLastAccountName()
            getAddedAssetsFromDb()
            getTransactionsFromDb()
            getAccountsFromDb()
            startReceivingData()
        }

        initViews()
        initFragments()
        initObservers()
    }

    private fun initFragments() = FindFragmentById.init()

    private fun initObservers(){
        viewModel.nextFragment.observe(this) {
            val fragment = FindFragmentById.getFragment(it)
            val transaction = supportFragmentManager.beginTransaction()
            transaction.replace(R.id.nav_host_fragment, fragment)
            transaction.commit()
        }

        viewModel.nightMode.observe(this){
            var isNightMode = false
            if (it == true){
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                binding.btnChangeAcc.setImageResource(R.drawable.ic_arrow_down)
                isNightMode = true
            }
            else {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                binding.btnChangeAcc.setImageResource(R.drawable.ic_arrow_down_night)
            }
            viewModel.setNightTheme(isNightMode)
        }

        viewModel.currentAccount.observe(this){
            binding.portfolioName.text = it.name
            viewModel.apply {
                getAddedAssetsFromDb()
                getTransactionsFromDb()
                getAccountsFromDb()
                setSavedAppTheme()
                createFragment(FindFragmentById.PORTFOLIO, FindFragmentById.HISTORY)
                openFragment(FindFragmentById.PORTFOLIO)
                calculateBalance()
            }
        }
    }

    private fun initViews(){
        binding.bottomNavMenu.setOnNavigationItemSelectedListener(BottomNavigationView.OnNavigationItemSelectedListener
        {
            val id = it.itemId
            val fragmentId: FindFragmentById =
                when (id) {
                    R.id.nav_history -> FindFragmentById.HISTORY
                    R.id.nav_portfolio -> FindFragmentById.PORTFOLIO
                    else -> FindFragmentById.SETTINGS
                }
            viewModel.openFragment(fragmentId)
            true
        })

        binding.btnChangeAcc.setOnClickListener{
            if (viewModel.nextFragment.value == FindFragmentById.ACCOUNT_MANAGEMENT)
                viewModel.openFragment(FindFragmentById.PORTFOLIO)
            else
                viewModel.openFragment(FindFragmentById.ACCOUNT_MANAGEMENT)
        }
    }

    override fun onBackPressed() {
        if (viewModel.nextFragment.value != FindFragmentById.PORTFOLIO)
            viewModel.openFragment(FindFragmentById.PORTFOLIO)
        else
            super.onBackPressed()
    }
}